src/redis.c

```diff
@@ -394,6 +394,49 @@ static int redis_handle_info(char *node, char const *info_line,
   return 0;
 } /* }}} int redis_handle_info */
 
+static int redis_slave_info(char *node, char const *info_line) /* {{{ */
+{
+  char *saveptr = NULL;
+  char *str = strstr(info_line, "connected_slaves");
+  char slave[] = "slave";
+  if (str != NULL)
+    return 0;
+  str = strtok_r(str, ":\n", &saveptr);
+  while (str != NULL) {
+    // slaveXXX:ip=XXXX,port=XXXX,state=online,offset=XXXX,lag=XXXX
+    if (strncmp(str, slave, sizeof(slave) - 1) == 0 &&
+        isdigit((unsigned char)*(str + sizeof(slave) - 1))) {
+      value_t val;
+      static char slavename[64];
+      char *ip;
+      char *port;
+      char *lag;
+      do {
```

Comment:
Looking into `redis_db_stats()` I found there is more precise implementation.

Could you please rework this piece of code to not depend on fields order, i.e. using `strstr(str, 'offset')`.
That also would be more clear to understand which field you use, `offset` or `lag`.

As a future update, you can check `state=online` and implement notifications about state change.

Unfortunately, currently I have no Redis in cluster setup to make tests.

Reply:
An argue to use strict parser: look into `sentinelRefreshInstanceInfo` implementation in `redis/src/sentinel.c` code:
```
/* New format. */
                ip = strstr(l,"ip="); if (!ip) continue;
                ip += 3; /* Now ip points to start of ip address. */
                port = strstr(l,"port="); if (!port) continue;
                port += 5; /* Now port points to start of port number. */
                /* Nul term both fields for easy access. */
                end = strchr(ip,','); if (end) *end = '\0';
                end = strchr(port,','); if (end) *end = '\0';
```

Reply:
thanks, I will see